### ISSUE TITLE
Bio::DB::BigFile fails some unit tests

### ISSUE THREAD

**Issue by GodloveD** [GMOD/GBrowse-Adaptors#16]
When installing Bio::DB::BigFile (as a dependency to PCAP-core) it fails some unit tests. Here is the build.log.

```
cpanm (App::cpanminus) 1.7042 on perl 5.018002 built for x86_64-linux-thread-multi-ld
Work directory is /home/godlovedc/.cpanm/work/1467299881.7444
You have make /usr/bin/make
You have LWP 6.15
You have /bin/tar: tar (GNU tar) 1.23
Copyright (C) 2010 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>.
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Written by John Gilmore and Jay Fenlason.
You have /usr/bin/unzip
Searching Bio::DB::BigFile () on cpanmetadb ...
--> Working on Bio::DB::BigFile
Fetching http://cpan.metacpan.org/authors/id/L/LD/LDS/Bio-BigFile-1.07.tar.gz
-> OK
Unpacking Bio-BigFile-1.07.tar.gz
Entering Bio-BigFile-1.07
Checking configure dependencies from META.json
Checking if you have Module::Build 0.38 ... Yes (0.4218)
Checking if you have ExtUtils::Install 1.46 ... Yes (2.04)
Configuring Bio-BigFile-1.07
Running Build.PL
Created MYMETA.yml and MYMETA.json
Creating new 'Build' script for 'Bio-BigFile' version '1.07'
-> OK
Checking dependencies from MYMETA.json ...
Checking if you have IO::String 0 ... Yes (1.08)
Checking if you have ExtUtils::CBuilder 0 ... Yes (0.280210)
Checking if you have Bio::Root::Version 1.005009001 ... Yes (1.006924)
Building and testing Bio-BigFile-1.07
Building Bio-BigFile
cc -I/usr/local/apps/ascatNgs/3.0.2/PCAP-core_git_repo/install_tmp/kent/src/inc -I/usr/local/Perl/5.18.2/lib/perl5/5.18.2/x86_64-linux-thread-multi-ld/CORE -DVERSION="1.07" -DXS_VERSION="1.07" -fPIC -Wno-format -Wformat=0 -DMACHTYPE_i686 -c -D_REENTRANT -D_GNU_SOURCE -fPIC -fno-strict-aliasing -pipe -fstack-protector -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -O2 -o lib/Bio/DB/BigFile.o lib/Bio/DB/BigFile.c
ExtUtils::Mkbootstrap::Mkbootstrap('blib/arch/auto/Bio/DB/BigFile/BigFile.bs')
cc -shared -O2 -L/usr/local/lib -fstack-protector -o blib/arch/auto/Bio/DB/BigFile/BigFile.so lib/Bio/DB/BigFile.o /usr/local/apps/ascatNgs/3.0.2/PCAP-core_git_repo/install_tmp/kent/src/lib/i686/jkweb.a -lz -lssl
# Test 20 got: "0.176919166116602751" (t/01.bigwig.t at line 89)
#    Expected: "0.176919166116602744"
#  t/01.bigwig.t line 89 is: ok ($extended_summary->sumData(1)/$extended_summary->validCount(1),$summary->[1]);
# Test 23 got: "0.0798449599229382158" (t/01.bigwig.t at line 94)
#    Expected: "0.079844959922938219"
#  t/01.bigwig.t line 94 is: ok($bin_sum->[300]->sumData/$bin_sum->[300]->validCount,$summary->[300]);
# Test 28 got: "0.0798449599229382158" (t/01.bigwig.t at line 101)
#    Expected: "0.079844959922938219"
#  t/01.bigwig.t line 101 is: ok($es_list->[300]{sumData}/$es_list->[300]{validCount},$summary->[300]);
t/01.bigwig.t .......
Failed 3/29 subtests
# Test 12 got: "1.00004166666666667" (t/02.bigbed.t at line 53)
#    Expected: "1.00004166666666672"
#  t/02.bigbed.t line 53 is: ok($es->[-2]{validCount}/(12_000_000/500),$summary->[-2]);
t/02.bigbed.t .......
Failed 1/15 subtests
t/03.bigwig_high.t .. ok
t/04.bigbed_high.t .. ok
t/05.bigwig_set.t ... ok

Test Summary Report
-------------------
t/01.bigwig.t     (Wstat: 0 Tests: 29 Failed: 3)
  Failed tests:  20, 23, 28
t/02.bigbed.t     (Wstat: 0 Tests: 15 Failed: 1)
  Failed test:  12
Files=5, Tests=135, 12 wallclock secs ( 0.07 usr  0.06 sys +  1.36 cusr  0.72 csys =  2.21 CPU)
Result: FAIL
Failed 2/5 test programs. 4/135 subtests failed.
-> FAIL Installing Bio::DB::BigFile failed. See /home/godlovedc/.cpanm/work/1467299881.7444/build.log for details. Retry with --force to force install it.
```

@keiranmraine thought it might be related to the way in which I installed nettle.  For the record, I was unable to use a package manager to install nettle and I used this method.

```
umask 002
ASCAT_DIR=/usr/local/apps/ascatNgs/3.0.2
mkdir -p $ASCAT_DIR
cd $ASCAT_DIR
wget https://git.lysator.liu.se/nettle/nettle/repository/archive.tar.gz?ref=nettle_3.2_release_20160128 -O nettle.tar.gz
mkdir -p nettle.3.2
tar --strip-components 1 -C nettle.3.2 -zxf nettle.tar.gz
cd nettle.3.2
./.bootstrap
./configure --prefix=$ASCAT_DIR/nettle --disable-pic --disable-shared
make
make check
# all passed
make install
cd ..
rm -fr nettle.*
```

**Comment by keiranmraine** [comment]
On further investigation it shouldn't be related to nettle.  I'm slightly concerned that this appears to be a floating point issue.  @GodloveD could you add the details of your system to the issue please, this set of command should give the needed info:

```
bash -c 'set -x; lsb_release -a; uname -a; sw_vers; system_profiler; grep MemTotal /proc/meminfo' >& system.info
```

**Comment by GodloveD** [comment]
Here is the output:
```
+ lsb_release -a
LSB Version: :base-4.0-amd64:base-4.0-noarch:core-4.0-amd64:core-4.0-noarch:graphics-4.0-amd64:graphics-4.0-noarch:printing-4.0-amd64:printing-4.0-noarch
Distributor ID: RedHatEnterpriseServer
Description:    Red Hat Enterprise Linux Server release 6.8 (Santiago)
Release:        6.8
Codename:       Santiago
+ uname -a
Linux biowulf.nih.gov 2.6.32-642.1.1.el6.x86_64 #1 SMP Fri May 6 14:54:05 EDT 2016 x86_64 x86_64 x86_64 GNU/Linux
+ sw_vers
bash: sw_vers: command not found
+ system_profiler
bash: system_profiler: command not found
+ grep MemTotal /proc/meminfo
MemTotal:       123883560 kB
```

**Comment by andrewyatz** [comment]
I've seen something similar on my new build of C libraries. Details of my system as follows:

```
+ lsb_release -a
LSB Version:    :core-4.1-amd64:core-4.1-noarch:cxx-4.1-amd64:cxx-4.1-noarch:desktop-4.1-amd64:desktop-4.1-noarch:languages-4.1-amd64:languages-4.1-noarch:printing-4.1-amd64:printing-4.1-noarch
Distributor ID: RedHatEnterpriseServer
Description:    Red Hat Enterprise Linux Server release 7.2 (Maipo)
Release:        7.2
Codename:       Maipo
+ uname -a
Linux ebi-006.ebi.ac.uk 3.10.0-327.10.1.el7.x86_64 #1 SMP Sat Jan 23 04:54:55 EST 2016 x86_64 x86_64 x86_64 GNU/Linux
+ sw_vers
bash: sw_vers: command not found
+ system_profiler
bash: system_profiler: command not found
+ grep MemTotal /proc/meminfo
MemTotal:       131824240 kB
```

Note that I didn't see this on the previous RHEL6 system. To be fair though assuming a floating point number is going to equal another one is a bit suspect to me. Better to either use `sprintf("%.6f", $val)` to round to 6 places (if we want that) or switch to `Math::Round` or `Math::BigFloat`.

**Comment by cjfields** [comment]
Fp matching tests are notoriously brittle if using perfect matching; keiranmraine's fixes in #20 should help this.
